src/main/java/uk/gov/hmcts/divorce/systemupdate/event/AdminUnlinkApplicant1FromCase.java

```diff
@@ -0,0 +1,57 @@
+package uk.gov.hmcts.divorce.systemupdate.event;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.sdk.api.CCDConfig;
+import uk.gov.hmcts.ccd.sdk.api.CaseDetails;
+import uk.gov.hmcts.ccd.sdk.api.ConfigBuilder;
+import uk.gov.hmcts.ccd.sdk.api.callback.AboutToStartOrSubmitResponse;
+import uk.gov.hmcts.divorce.divorcecase.model.CaseData;
+import uk.gov.hmcts.divorce.divorcecase.model.State;
+import uk.gov.hmcts.divorce.divorcecase.model.UserRole;
+import uk.gov.hmcts.divorce.solicitor.service.CcdAccessService;
+
+import java.util.List;
+
+import static uk.gov.hmcts.divorce.divorcecase.model.State.STATES_NOT_WITHDRAWN_OR_REJECTED;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.CREATOR;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.SUPER_USER;
+import static uk.gov.hmcts.divorce.divorcecase.model.access.Permissions.CREATE_READ_UPDATE;
+
+@Slf4j
+@Component
+public class AdminUnlinkApplicant1FromCase implements CCDConfig<CaseData, State, UserRole> {
+
+    @Autowired
+    private CcdAccessService ccdAccessService;
+
+    public static final String ADMIN_UNLINK_APPLICANT_1 = "admin-unlink-applicant1";
+
+    @Override
+    public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+        if (Boolean.parseBoolean(System.getenv().get("ADMIN_UNLINK_APPLICANT_1_ENABLED"))) {
+            configBuilder
+                .event(ADMIN_UNLINK_APPLICANT_1)
+                .forStates(STATES_NOT_WITHDRAWN_OR_REJECTED)
+                .name("Unlink Applicant1 from case")
+                .grant(CREATE_READ_UPDATE, SUPER_USER)
+                .retries(120, 120)
+                .aboutToSubmitCallback(this::aboutToSubmit);
+        }
+    }
+
+    public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(CaseDetails<CaseData, State> details,
+                                                                       CaseDetails<CaseData, State> beforeDetails) {
+
+        CaseData caseData = details.getData();
+
+
+        log.info("Admin unlinking Applicant1/Creator from case (id: {})",  details.getId());
```

Comment:
technically this is only [CREATOR] right now. Do you want to extend to applicant 1 user role as well?

Reply:
I updated the log message to simply include Creator, I guess we can leave it as CREATOR, as for this ticket as you mentioned we know the user is the CREATOR. If you were to extend to applicant 1 user role, do we then need to add a new user role (APPLICANT_1) in the UserRole.java class? As currently there are CITIZEN, CREATOR, APPLICANT_2 in this class.